frontend/src/app/workflows/workflows.service.ts

```diff
@@ -0,0 +1,64 @@
+import {Injectable} from '@angular/core';
+import {HttpClient, HttpHeaders} from '@angular/common/http';
+
+import {Observable} from 'rxjs';
+import {saveAs} from 'file-saver';
+import {environment} from '../../environments/environment';
+import {map} from 'rxjs/operators';
+import {GenerateResult} from './model/GenerateResult';
+import {createMapFromAnswer} from './model/helper/map.helper';
+import {ExportOptions} from './model/ExportOptions';
+
+@Injectable()
+export class WorkflowsService {
+
+  constructor(
+    private http: HttpClient,
+  ) {
+  }
+
+  public generate(data: string): Observable<GenerateResult> {
+    const url = environment.workflowsUrl + '/generate';
+    return this.http.post<any>(url, data).pipe(map(result => WorkflowsService.toGenerateResult(result)));
+  }
+
+  public downloadZip(cmContent: string, options: ExportOptions) {
+    const url = environment.workflowsUrl + '/download';
+    const headers: HttpHeaders = new HttpHeaders();
+
+    this.http.post(url, cmContent, {
+      headers: headers,
+      params: {
+        yaml: options.yaml,
+        board: options.board,
+        pages: options.pages,
+        objects: options.objects,
+        class: options.class,
+        fxmls: options.fxmls,
+      },
```

Comment:
Does `params: options` not work?

Reply:
I get the following error when using `params: options` (the reason why i used any in the previous iteration)
![image](https://user-images.githubusercontent.com/39198379/153931156-e0a1a8e3-4d6a-4aab-bbfa-b5d51684f6c9.png)

Reply:
Ah yes, I remember that I ran into the same issue :sweat_smile: Nevermind then :)